605/theme updates

JaeGif opened the pull request:
## This PR:
Resolves #605
 
- theme.js updated with new color scheme
- added 2 new fonts to theme.js
- color scheme applied to major site elements
- started adding/changing elements to match the Figma board on Profile page
- started adding/changing button elements to match Figma board on Contacts page

## Future Steps/PRs Needed to Finish This Work (optional):
This PR is just the beginning of the work that needs to be done to overhaul the UI. With the addition of new themes and fonts, the next steps can be broken into multiple issues. I'd recommend making an issue per page to be re-themed since most pages require the addition of new elements/functionality and rewriting unit tests. Some components (like the DocumentList component) will need major overhauls and might warrant their own issue.

Also should contact the UI team and see if there's any more/other designs incoming before we get too far on that.

JaeGif committed:
add new color palettes to theme.js

JaeGif committed:
add new font families

JaeGif committed:
theme: share docs add docs button

JaeGif committed:
add profile styling

JaeGif committed:
begins contacts theme

JaeGif committed:
new colors add to theme

JaeGif committed:
fixing test 1

JaeGif committed:
add home page theming

JaeGif committed:
add home page themeing

JaeGif committed:
temp fix breaking tests

JaeGif committed:
Merge branch 'Development' into 605/theme_updates

leekahung commented on src/components/Modals/UploadDocumentModal.jsx:
```diff
@@ -133,6 +139,7 @@ const UploadDocumentModal = ({ showModal, setShowModal }) => {
         onConfirm={confirmationModalType === 'add' ? handleDocAdd : handleDocReplace}
         uploadType={confirmationModalType}
       />
+      km
```
There's an additional `km` string in the modal (see upper left)

![Screenshot 2024-04-29 at 11 30 08 AM](https://github.com/codeforpdx/PASS/assets/14917816/70b19777-868b-43aa-8645-dcc93b281fcf)

leekahung commented on src/components/Modals/UploadButtonGroup.jsx:
```diff
@@ -91,27 +92,37 @@ const UploadButtonGroup = ({ file, setFile }) => {
-          accept=".pdf, .docx, .doc, .txt, .rtf, .gif, .png, .jpeg, .jpg, .webp"
-        />
+        Upload File
+        <input type="file" hidden accept=".pdf, .docx, .doc, .txt, .rtf" />
```
There could be files users wish to update that are not text/pdf files, but also image files. That was why we had additional extensions like png and jpg accepted as acceptable file inputs.

JaeGif replied:
I changed this one just because the Figma design has a separate button for media files

leekahung replied:
Think we should add the new button in too if that's the case since there would be times when people just upload images. It might look a little funky though since it's no longer 2 over 2, but 3 over 2.

JaeGif replied:
Gotcha, added the upload photos button back in and changed the layout to column for now so it doesn't look too funky. There's a unit test that counts the number of buttons rendered, so I updated it to count 3 now.

leekahung replied:
Sounds good, thanks!

leekahung commented on src/components/Modals/UploadDocumentModal.jsx:
```diff
@@ -17,14 +17,19 @@ import { DatePicker } from '@mui/x-date-pickers/DatePicker';
 import useNotification from '../../hooks/useNotification';
 import UploadDocumentConfirmationModal from './UploadDocumentConfirmationModal';
-
+/* 
```
Is this `Transition` component gonna be used for an up coming PR for modal/drawer animation?

JaeGif replied:
<img width="265" alt="Screenshot 2024-04-29 at 11 40 59 AM" src="https://github.com/codeforpdx/PASS/assets/105343665/7d85abdb-2678-40af-8b3b-90c5f6422885">

Yeah, the button grouping, changing the file type, icons, transition etc are first steps toward achieving this ^ design. I started these before the scope of the issue changed.

leekahung commented on src/components/Modals/UploadButtonGroup.jsx:
```diff
@@ -91,27 +92,37 @@ const UploadButtonGroup = ({ file, setFile }) => {
+        <input type="file" hidden accept=".pdf, .docx, .doc, .txt, .rtf" />
       </Button>
 
+      {/*       <Button
```
A bit confused as to why the buttons were split up in this PR?

leekahung commented on src/pages/Contacts.jsx:
```diff
@@ -76,15 +79,42 @@ const Contacts = () => {
-        </Button>
+        <Box sx={{ display: 'flex', justifyContent: 'space-between' }}>
+          <FormControl sx={{ minWidth: 120 }} size="small">
+            <Select
```
I believe this `Select` field is here for mobile and not the desktop version?

Since we'll no longer have access to the MUI table's sorting capabilities when we switch to cards, would make sense to have that we'll need one for a custom sorting function in follow-up PRs.

![Screenshot 2024-04-29 at 11 40 03 AM](https://github.com/codeforpdx/PASS/assets/14917816/d4276c02-8ee6-4b4e-a33a-d508626588e7)

JaeGif replied:
Do you mean it's only supposed to be there for a mobile version?

leekahung replied:
Think we technically could have the sorting trigger on Desktop as well from the `Select` field, but that might be duplicate functionality?

JaeGif replied:
> Think we technically could have the sorting trigger on Desktop as well from the `Select` field, but that might be duplicate functionality?

I'm not sure I understand what you mean. As far as I'm aware, there's only mobile wireframes on Figma. Do you mean that the Select input sorting should only be available in a mobile view?

leekahung replied:
Yes, the `Select` sorting should only be available in Mobile.

JaeGif committed:
remove accidental 'km' string

leekahung commented:
Related to this PR, there still a bit of theming we could update from the Inbox/Outbox buttons, namely the custom styling from React Paginate (see `MessageFolder.jsx` for the component and `PaginationStyles.js` for the custom styles and classes used).

![Screenshot 2024-04-29 at 11 46 36 AM (2)](https://github.com/codeforpdx/PASS/assets/14917816/36859c60-e54c-4828-9819-c32a37d048af)

I believe Ben at the time made it so that the styling and coloring were in line with what we had before. Think we could do the same thing here with the new theme.

leekahung reviewed (changes requested):
Alright, think that touches most of the things that still needed to be addressed. But outside of those, the changes made with the new theme looks great!

JaeGif committed:
add media button back, change layout to column

JaeGif committed:
fix spelling change that caused test to break

JaeGif committed:
add consistent theming to messaging inbox/outbox

JaeGif committed:
select only available in small breakpoint

JaeGif commented:
> Alright, think that touches most of the things that still needed to be addressed. But outside of those, the changes made with the new theme looks great!

Great, I think I've addressed most of the points in the last few commits. Please let me know if there's anything else or if something doesn't fit the requirements. Cheers!

leekahung reviewed (approved):
Looking great! Approving.
